Intersect.Client/Entities/Player.cs

```diff
@@ -2383,32 +2383,13 @@ protected override bool ShouldDrawHpBar
                     return false;
                 }
 
-                if (IsHovered)
-                {
-                    return true;
-                }
-
-                if (Id == Globals.Me.Id)
-                {
-                    return Globals.Database.MyOverheadHpBar;
-                }
-
-                if (Globals.Database.PartyMemberOverheadHpBar && Globals.Me.IsInMyParty(this))
-                {
-                    return true;
-                }
-
-                if (Globals.Database.FriendOverheadHpBar && Globals.Me.IsFriend(this))
-                {
-                    return true;
-                }
-
-                if (Globals.Database.GuildMemberOverheadHpBar && Globals.Me.IsGuildMate(this))
-                {
-                    return true;
-                }
-
-                return Globals.Database.PlayerOverheadHpBar;
+                var me = Globals.Me;
+                bool userPreference = (Globals.Database.MyOverheadHpBar && Id == me.Id) ||
+                                      (Globals.Database.PlayerOverheadHpBar && Id != me.Id) ||
+                                      (Globals.Database.PartyMemberOverheadHpBar && me.IsInMyParty(this)) ||
+                                      (Globals.Database.FriendOverheadHpBar && me.IsFriend(this)) ||
+                                      (Globals.Database.GuildMemberOverheadHpBar && me.IsGuildMate(this));
+                return userPreference || IsHovered;
```

Comment:
Revert this. This is not more readable, and forces expensive operations to happen before IsHovered which is a much less costly check.

The original structure and order was very intentional.